src/Streamly/Internal/Memory/Array/Types.hs

```diff
@@ -1147,22 +1145,20 @@ lpackArraysChunksOf n (Fold step1 initial1 extract1) =
             -- XXX we can pass the module string from the higher level API
             error $ "Streamly.Internal.Memory.Array.Types.packArraysChunksOf: the size of "
                  ++ "arrays [" ++ show n ++ "] must be a natural number"
-        r1 <- initial1
-        return (Tuple' Nothing r1)
+        Tuple' Nothing <$> initial1
```

Comment:
@harendra-kumar Do you guys feels this is better than the original one?

Reply:
I prefer the original one, its easier to understand. Can we shut up hlint for this?

Reply:
However, at some places we can do with functor instead of using monad operations. In theory, in some cases functor may be more efficient compared to monad depending on the implementation. But I am not sure if there are any cases like that. If we shut it up will it preclude any such suggestions?

Reply:
I don't know about any such suggestions, but in some cases like here it definitely feels better
```diff
    {-# INLINE readPrec #-}
-    readPrec = do
-          xs <- readPrec
-          return (fromList xs)
+    readPrec = fromList <$> readPrec
```
to write it this way, so there isn't a rule probably to stop it only in that kind of code. We have to either accept this or not do this anywhere else in the code.

Reply:
Yes, that's what I was talking about.

Reply:
It may be annoying if there are too many such cases where we prefer the original code. Maybe we can shut it up for now and write a comment in the hlint config that this can be useful in some cases. So may be once in a while or in dev builds we can enable it and correct the ones we like.

Reply:
I think we'll just have to play fast and loose with the suggestions. I'll be merging this with the `Use <$>`  suggestion removed from a few cases.